src/Features/LanguageServer/Protocol/Extensions/ProtocolConversions.cs

```diff
@@ -199,6 +199,22 @@ public static Uri CreateAbsoluteUri(string absolutePath)
             }
         }
 
+        internal static Uri CreateRelativePatternBasUri(string path)
+        {
+            // According to VSCode LSP RelativePattern spec, 
+            // found at https://github.com/microsoft/vscode/blob/9e1974682eb84eebb073d4ae775bad1738c281f6/src/vscode-dts/vscode.d.ts#L2226
+            // the baseUri should not end in a trailing separator, nor should it 
+            // have any relative segmeents (., ..)
+            if (path[^1] == System.IO.Path.DirectorySeparatorChar)
```

Comment:
every time we hand write some file path code, i die inside. :)

Reply:
Me too... me too

Reply:
Path.TrimEnd ?

Reply:
get out of here with your sense and logic. TBH I was just trying to avoid any allocations but I'm guessing trimend will be efficient enough